### ISSUE TITLE
Not working on, don't know why

### ISSUE THREAD

**Issue by Nemo7CZ** [undertherain/furiganize#3]
I'm not sure what I'm doing wrong:

Fedora 40
Libreoffice 24.2.6.2

installed packages:
kakasi 2.3.6-28.fc40
mecab 0.996-7.fc40
mecab-jumandic-EUCJP 5.1.20070304-30.fc40

![Error message](https://github.com/user-attachments/assets/feccecb8-2dc2-4ce7-b363-2ad1157147bb)

**Comment by undertherain** [comment]
Hi!
try installing the dictionary that is specified in README - mecab-naist-jdic-eucjp

**Comment by undertherain** [comment]
also, can you give an example of  text on which it fails?

**Comment by Nemo7CZ** [comment]
I can't find the "naist" packgage:

> $ dnf search mecab
Last metadata expiration check: 0:00:20 ago on So 19. října 2024, 09:25:41.
========================= Name Exactly Matched: mecab ==========================
mecab.x86_64 : Yet Another Part-of-Speech and Morphological Analyzer
mecab.i686 : Yet Another Part-of-Speech and Morphological Analyzer
======================== Name & Summary Matched: mecab =========================
groonga-plugin-tokenizer-mecab.x86_64 : MeCab tokenizer for Groonga
mecab-devel.x86_64 : Libraries and Header files for Mecab
mecab-devel.i686 : Libraries and Header files for Mecab
mecab-ipadic.x86_64 : IPA dictionary for MeCab
mecab-ipadic-EUCJP.x86_64 : IPA dictionary for Mecab with encoded by EUC-JP
mecab-java.x86_64 : Java binding for MeCab
mecab-jumandic.x86_64 : JUMAN dictorionary for MeCab
mecab-jumandic-EUCJP.x86_64 : JUMAN dictionary for Mecab with encoded by EUC-JP
perl-mecab.x86_64 : Perl binding for MeCab
ruby-mecab.x86_64 : Ruby binding for MeCab


I've now tried installing mecab-devel and theipadic and ipadic-EUCJP ones but it still fails. (yestarday in a different way, today the same):
When I try to furiganize a single word, it does the same thing as shown in the first message.
(Highlighted a word and ran it.)
![Screenshot from 2024-10-19 09-35-46](https://github.com/user-attachments/assets/eeef9754-1899-4bc9-ac8a-680ac942d8e9)


Console keeps repeating this:
> sequence '\['
>   (kanji, reading) = re.match("(.+)\[(.*)\]", node).groups()
> <unknown>:271: SyntaxWarning: invalid escape sequence '\['
> <unknown>:271: SyntaxWarning: invalid escape sequence '\['
> <unknown>:271: SyntaxWarning: invalid escape sequence '\['
> <unknown>:271: SyntaxWarning: invalid escape sequence '\['

**Comment by Nemo7CZ** [comment]
Is there a way to install mecab-naist-jdic-eucjp manually?

**Comment by undertherain** [comment]
which Python version do you use?

**Comment by undertherain** [comment]
![image](https://github.com/user-attachments/assets/6a1e1b88-3cfb-40cc-b416-1ddf7f1c991a)
I am not sure it is because of a different dict, but that's the first obvious difference that stood out

**Comment by Nemo7CZ** [comment]
> ![image](https://private-user-images.githubusercontent.com/1635907/378079364-6a1e1b88-3cfb-40cc-b416-1ddf7f1c991a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkzNDAzNzEsIm5iZiI6MTcyOTM0MDA3MSwicGF0aCI6Ii8xNjM1OTA3LzM3ODA3OTM2NC02YTFlMWI4OC0zY2ZiLTQwY2MtYjQxNi0xZGRmN2YxYzk5MWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTlUMTIxNDMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODdjMWRmYzYyZmUzOWY0OTY3OTgwZjZlNDYzNDZkZjcyMjY4Mjc3YzI0NWNmMDQ4OTJlZGQ0Yjc5MGQwMGZiNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.3vXjT6ATlgmLTitrqIe_EpyjMy3M2pejiLY4XGjHFhs) I am not sure it is because of a different dict, but that's the first obvious difference that stood out

Yes, is there a way to install the recommended one manually? It's not available in the Fedora repos.


> which Python version do you use?

I have the default: Python 3.12.6.

**Comment by undertherain** [comment]
If you just run the script from the command line it should do some tests
can you try `python3  macro_furiganize.py` ?

**Comment by undertherain** [comment]
I got it working on Fedora with following packages:
`kakasi kakasi-dict mecab mecab-ipadic-EUCJP`
